File: backend/tests/fixtures/ef_models/Post.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MyApp.Models
{
    /// <summary>
    /// Post entity representing a blog post authored by a user.
    /// </summary>
    [Table("Posts")]
    public class Post
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(200)]
        public string Title { get; set; }

        public string Content { get; set; }

        [Required]
        public bool IsPublished { get; set; }

        public int ViewCount { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime? UpdatedAt { get; set; }

        [ForeignKey("Author")]
        public int AuthorId { get; set; }

        // Navigation — many-to-one back to User
        public User Author { get; set; }

        // Navigation — one-to-many to comments
        public ICollection<Comment> Comments { get; set; }

        [Column("category_id")]
        public int? CategoryId { get; set; }
    }
}

File: backend/tests/fixtures/jpa_models/Post.java
package com.example.models;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.List;

/**
 * Post entity representing a blog post authored by a user.
 */
@Entity
@Table(name = "posts")
public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "title", nullable = false, length = 200)
    private String title;

    @Column(name = "content", columnDefinition = "TEXT")
    private String content;

    @Column(name = "published")
    private Boolean published;

    @Column(name = "view_count")
    private Integer viewCount;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id", nullable = false)
    private User author;

    @OneToMany(mappedBy = "post", cascade = CascadeType.ALL)
    private List<Comment> comments;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "metadata_id", referencedColumnName = "id")
    private PostMetadata metadata;

    public Post() {}

    public Long getId() { return id; }
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
}
